rxnetty-http/src/main/java/io/reactivex/netty/protocol/http/client/HttpClientRequest.java

```diff
@@ -79,9 +79,9 @@ protected HttpClientRequest(OnSubscribe<HttpClientResponse<O>> onSubscribe) {
      *
      * @param contentSource Content source for the request.
      *
-     * @return An new instance of {@link Observable} which can be subscribed to execute the request.
+     * @return An new instance of the {@link HttpClientRequest} which can be subscribed to execute the request.
      */
-    public abstract Observable<HttpClientResponse<O>> writeContent(Observable<I> contentSource);
```

Comment:
Apologies that I did not remember this before in our offline conversations.

The reason why `write*()` methods return `Observable<HttpClientResponse<O>>` is that the request does not support writing multiple content streams. If we return `HttpClientRequest` here, a subsequent `write*()` call will overwrite the previously written content.

This method signature avoids such unintended mistakes.